### ISSUE TITLE
Must use import to load ES Module

### ISSUE THREAD

**Issue by nnscr** [aurelia/aurelia#787]
# 🐛 Bug Report

When starting a freshly created project I get the error "Must use import to load ES Module"

## 😯 Current Behavior

```
ℹ ｢wds｣: Project is running at http://localhost:9000/
ℹ ｢wds｣: webpack output is served from /
ℹ ｢wds｣: Content not from webpack is served from [...]
ℹ ｢wds｣: 404s will fallback to /index.html
(node:349883) Warning: require() of ES modules is not supported.
require() of [...]/node_modules/@aurelia/webpack-loader/dist/umd/index.js from [...]/node_modules/loader-runner/lib/loadLoader.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
Instead rename index.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from [...]/node_modules/@aurelia/webpack-loader/package.json.
ℹ ｢wdm｣: wait until bundle finished: /
✖ ｢wdm｣: Hash: d9e950c12d5ada749559
Version: webpack 4.41.3
Time: 1395ms
Built at: 16.12.2019 13:14:19
          Asset       Size  Chunks             Chunk Names
entry-bundle.js    876 KiB    main  [emitted]  main
     index.html  285 bytes          [emitted]  
Entrypoint main = entry-bundle.js
[0] multi (webpack)-dev-server/client?http://localhost:9000 ./src/main.ts 40 bytes {main} [built]
[./node_modules/ansi-html/index.js] 4.16 KiB {main} [built]
[./node_modules/ansi-regex/index.js] 135 bytes {main} [built]
[./node_modules/html-entities/index.js] 231 bytes {main} [built]
[./node_modules/loglevel/lib/loglevel.js] 8.36 KiB {main} [built]
[./node_modules/strip-ansi/index.js] 161 bytes {main} [built]
[./node_modules/webpack-dev-server/client/index.js?http://localhost:9000] (webpack)-dev-server/client?http://localhost:9000 4.29 KiB {main} [built]
[./node_modules/webpack-dev-server/client/overlay.js] (webpack)-dev-server/client/overlay.js 3.51 KiB {main} [built]
[./node_modules/webpack-dev-server/client/socket.js] (webpack)-dev-server/client/socket.js 1.53 KiB {main} [built]
[./node_modules/webpack-dev-server/client/utils/createSocketUrl.js] (webpack)-dev-server/client/utils/createSocketUrl.js 2.89 KiB {main} [built]
[./node_modules/webpack-dev-server/client/utils/log.js] (webpack)-dev-server/client/utils/log.js 964 bytes {main} [built]
[./node_modules/webpack-dev-server/client/utils/reloadApp.js] (webpack)-dev-server/client/utils/reloadApp.js 1.59 KiB {main} [built]
[./node_modules/webpack-dev-server/client/utils/sendMessage.js] (webpack)-dev-server/client/utils/sendMessage.js 402 bytes {main} [built]
[./node_modules/webpack/hot sync ^\.\/log$] (webpack)/hot sync nonrecursive ^\.\/log$ 170 bytes {main} [built]
[./src/main.ts] 1.19 KiB {main} [not cacheable] [built] [failed] [1 error]
    + 18 hidden modules

ERROR in ./src/main.ts
Module build failed (from ./node_modules/@aurelia/webpack-loader/dist/umd/index.js):
Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: [...]/node_modules/@aurelia/webpack-loader/dist/umd/index.js
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1156:13)
    at Module.load (internal/modules/cjs/loader.js:976:32)
    at Function.Module._load (internal/modules/cjs/loader.js:884:14)
    at Module.require (internal/modules/cjs/loader.js:1016:19)
    at require (internal/modules/cjs/helpers.js:69:18)
    at loadLoader ([...]/node_modules/loader-runner/lib/loadLoader.js:18:17)
    at iteratePitchingLoaders ([...]/node_modules/loader-runner/lib/LoaderRunner.js:169:2)
    at iteratePitchingLoaders ([...]/node_modules/loader-runner/lib/LoaderRunner.js:165:10)
    at [...]/node_modules/loader-runner/lib/LoaderRunner.js:176:18
    at loadLoader ([...]/node_modules/loader-runner/lib/loadLoader.js:47:3)
Child html-webpack-plugin for "index.html":
     1 asset
    Entrypoint undefined = index.html
    [./node_modules/html-webpack-plugin/lib/loader.js!./index.ejs] 431 bytes {0} [built]
    [./node_modules/lodash/lodash.js] 528 KiB {0} [built]
    [./node_modules/webpack/buildin/global.js] (webpack)/buildin/global.js 472 bytes {0} [built]
    [./node_modules/webpack/buildin/module.js] (webpack)/buildin/module.js 497 bytes {0} [built]
ℹ ｢wdm｣: Failed to compile.
```

## 🌍 Your Environment

To reproduce:
` npx makes aurelia new-project-name -s typescript,css-module,sass,no-unit-tests`
` npm start`

(Note: same issue with shadow dom instead of css-module)

| Software         | Version(s) |
| ---------------- | ---------- |
| Aurelia          | dev
| Language         | -
| Browser          | -
| Bundler          | Webpack 4.0
| Operating System | Arch Linux
| NPM/Node/Yarn   | NPM 6.12.1, Node 13.2.0

package.json:
```
  "dependencies": {
    "aurelia": "dev"
  },
  "devDependencies": {
    "htmlhint": "^0.11.0",
    "sass-lint": "^1.0.0",
    "rimraf": "^2.0.0",
    "webpack": "^4.0.0",
    "webpack-cli": "^3.0.0",
    "webpack-dev-server": "^3.0.0",
    "html-webpack-plugin": "^3.0.0",
    "style-loader": "^1.0.0",
    "css-loader": "^3.0.0",
    "@aurelia/webpack-loader": "dev",
    "ts-loader": "^6.0.0",
    "sass-loader": "^8.0.0",
    "node-sass": "^4.0.0",
    "postcss-loader": "^3.0.0",
    "autoprefixer": "^9.0.0",
    "@types/node": "^12.0.0",
    "typescript": "^3.0.0",
    "tslint": "^5.0.0"
  },
```

**Comment by fkleuver** [comment]
Thanks for reporting this.

@3cp We ran into this before. Do you remember what the problem/solution was?

Also I thought we had some automated e2e tests creating cli projects or do we not yet have those?

**Comment by 3cp** [comment]
This one is new.

The log said it clear: `or remove "type": "module" from [...]/node_modules/@aurelia/webpack-loader/package.json`.

Because of https://github.com/aurelia/aurelia/commit/4338eca353

This is wrong, we cannot just turn on `"type":"module"`.

1. our `main` is still pointing to umd build, **those js files are not in esm format**. That's the complain in this issue.

2. even if we remove umd build, then point `main` to esm build, **it still would not work**. Because our esm build is generated by TS, there are lots of relative imports like `import {Foo} from './foo';`. This import statement is **invalid** according to nodejs esm spec. **nodejs** wants `import {Foo} from './foo.js';` (remember I told you this odd spec?).

3. there is no way we can just turn on esm in latest nodejs (either by `"type":"module"`, or with explicit `.cjs` and `.mjs` files) and still keep compatibility with old nodejs versions.

**In short, nodejs esm support is a bloody mess, don't use it now. It's a design from perfectionist ignoring practical pain.**

Regarding e2e tests, we have that in both skeleton repo https://github.com/aurelia/new, and this monorepo. The e2e tests in skeleton repo takes very very long time, so it's not turned on in CI. The e2e tests in this monorepo are from those examples apps, the problem is all examples apps are old ones without using conventions.

Once you rollback the `"type":"module"` commit, I can have to update the webpack example app to use conventions. (BTW, we don't have conventions plugins for parcel and browserify yet)

**Comment by 3cp** [comment]
Reference https://nodejs.org/dist/latest-v13.x/docs/api/esm.html#esm_dual_commonjs_es_module_packages

None of the official solution works for nodejs v8 or v10, the so called "solution" is only for nodejs v13.

**Comment by fkleuver** [comment]
It's the early design decision of nodejs to allow omitting the file extension that never made any sense imo. Perhaps we should just change everything to using explicit file extensions in the repo. That's the spec compliant way anyway. Native modules work this way in the browser too.

**Comment by 3cp** [comment]
In ts source code, we write `from './foo.ts'`, have to ask tsc to keep ext but change to `from './foo.js'`.

BTW, there is no way to turn on `"type":"module"` and support both nodejs v8 and v13 at same time. It's not a choice, have to rollback.

**Comment by fkleuver** [comment]
@nnscr This should be fixed now. Would you try again? :)

**Comment by 3cp** [comment]
I checked, it worked for me.

**Comment by nnscr** [comment]
@fkleuver Yep, works for me as well. Thank you!
